config.toml

```diff
@@ -3,129 +3,142 @@
 
-# Mention extractor will check spans up to max_span_len words
-# The default value is chosen to be big enough to hold any dev data span
-max_span_len = 64
+        # Controls how many pairs should be preserved per mention
+        # after applying rough scoring.
+        rough_k = 50
 
+    [DEFAULT.training_params]
+        # Training settings ==================
+        # The device where everything is to be placed. "cuda:N"/"cpu" are supported.
+        device = "cuda"
 
-# Pruning settings ===================
+        # Controls whether to fine-tune bert_model
+        bert_finetune = true
 
-# Controls how many pairs should be preserved per mention
-# after applying rough scoring.
-rough_k = 50
+        # Controls the dropout rate throughout all models
+        dropout_rate = 0.3
 
+        # Bert learning rate (only used if bert_finetune is set)
+        bert_learning_rate = 1e-5
 
-# Training settings ==================
+        # Task learning rate
+        learning_rate = 3e-4
 
-# Controls whether to fine-tune bert_model
-bert_finetune = true
+        # For how many epochs the training is done
+        train_epochs = 20
 
-# Controls the dropout rate throughout all models
-dropout_rate = 0.3
+        # Controls the weight of binary cross entropy loss added to nlml loss
+        bce_loss_weight = 0.5
 
-# Bert learning rate (only used if bert_finetune is set)
-bert_learning_rate = 1e-5
+        # The directory that will contain conll prediction files
+        conll_log_dir = "data/conll_logs"
 
-# Task learning rate
-learning_rate = 3e-4
 
-# For how many epochs the training is done
-train_epochs = 20
+    # =============================================================================
+    # Extra keyword arguments to be passed to bert tokenizers of specified models
+    [DEFAULT.tokenizer_kwargs]
+        [DEFAULT.tokenizer_kwargs.roberta-large]
+            "add_prefix_space" = true
 
-# Controls the weight of binary cross entropy loss added to nlml loss
-bce_loss_weight = 0.5
+        [DEFAULT.tokenizer_kwargs.spanbert-large-cased]
+            "do_lower_case" = false
 
-# The directory that will contain conll prediction files
-conll_log_dir = "data/conll_logs"
-
-# =============================================================================
-# Extra keyword arguments to be passed to bert tokenizers of specified models
-[DEFAULT.tokenizer_kwargs]
-    [DEFAULT.tokenizer_kwargs.roberta-large]
-        "add_prefix_space" = true
-
-    [DEFAULT.tokenizer_kwargs.spanbert-large-cased]
-        "do_lower_case" = false
-
-    [DEFAULT.tokenizer_kwargs.bert-large-cased]
-        "do_lower_case" = false
+        [DEFAULT.tokenizer_kwargs.bert-large-cased]
+            "do_lower_case" = false
 
 # =============================================================================
 # The sections listed here do not need to make use of all config variables
 # If a variable is omitted, its default value will be used instead
 
```

Comment:
I would suggest to add a parent section called, e.g., `language_models` or `encoders` to keep all of those under the same roof. Otherwise, they are all on same level as `DEFAULT`.

Reply:
you mean `.toml` doc or `Config` dataclass? 

We can do smth like:
```
[DEFAULT.encoder_params]
        [DEFAULT.encoder_params.tokenizer_kwargs]
            [DEFAULT.encoder_params.tokenizer_kwargs.roberta-large]
                "add_prefix_space" = true

            [DEFAULT.encoder_params.tokenizer_kwargs.spanbert-large-cased]
                "do_lower_case" = false

            [DEFAULT.encoder_params.tokenizer_kwargs.bert-large-cased]
                "do_lower_case" = false
```

and do the same indention in `Config` dataclass where `encoder_params` will be another dataclass

Reply:
@salisaresama  please let me know on that

Reply:
The refactor will be quite complicated for that. I will create a low prio task in backlog. For now assuming as good enough and merging